src/identity_tree/mod.rs

```diff
@@ -561,14 +560,9 @@ impl TreeState {
 
     #[must_use]
     pub fn get_proof_for(&self, item: &TreeItem) -> (Field, InclusionProof) {
-        let (leaf, root, proof) = match item.status {
-            ProcessedStatus::Pending => self.latest.get_leaf_and_proof(item.leaf_index),
-            ProcessedStatus::Processed => self.processed.get_leaf_and_proof(item.leaf_index),
-            ProcessedStatus::Mined => self.mined.get_leaf_and_proof(item.leaf_index),
-        };
+        let (leaf, root, proof) = self.latest.get_leaf_and_proof(item.leaf_index);
```

Comment:
Eventually we'll also want to remove the different tree versions. But let's do it in another PR

Reply:
For first PR I removed processed and mined tree (but keeping the status field in database).